docs/architecture/deep-dives/state/index.md

```diff
@@ -142,212 +143,192 @@ own key definition.
+:::warning
 
-type ExpectedAccountState = { myUserData: string };
+`ActiveUserState` has race condition problems. Do not use it for updates and consider transitioning
+your code to SingleUserState instead. [Read more.](#should-i-use-activeuserstate)
 
-const MY_GLOBAL_KEY_DEFINITION: KeyDefinitionLike = {
-  stateDefinition: { name: "myState" },
-  key: "myGlobalKey",
-};
-const MY_USER_KEY_DEFINITION: KeyDefinitionLike = {
-  stateDefinition: { name: "myState" },
-  key: "myUserKey",
-};
+:::
 
-export class MoveToStateProvider extends Migrator<10, 11> {
-  async migrate(migrationHelper: MigrationHelper): Promise<void> {
-    const existingGlobalData = await migrationHelper.get<ExpectedGlobalState>("global");
+`ActiveUserState<T>` is an object to help you maintain and view the state of the currently active
+user. If the currently-active user changes, like through account switching, the data this object
+represents will change along with it.
 
-    await migrationHelper.setGlobal(MY_GLOBAL_KEY_DEFINITION, {
-      myGlobalData: existingGlobalData.myGlobalData,
-    });
+### Updating state with `update`
 
-    const updateAccount = async (userId: string, account: ExpectedAccountState) => {
-      await migrationHelper.setUser(MY_USER_KEY_DEFINITION, {
-        myUserData: account.myUserData,
-      });
-    };
+The update method has options defined as follows:
 
-    const accounts = await migrationHelper.getAccounts<ExpectedAccountState>();
+```typescript
+{ActiveUser|SingleUser|Global}State<T> {
+  // ... rest of type left out for brevity
+  update<TCombine>(updateState: (state: T, dependency: TCombine) => T, options?: StateUpdateOptions);
+}
 
-    await Promise.all(accounts.map(({ userId, account }) => updateAccount(userId, account)));
-  }
+type StateUpdateOptions = {
+  shouldUpdate?: (state: T, dependency: TCombine) => boolean;
+  combineLatestWith?: Observable<TCombine>;
+  msTimeout?: number
 }
 ```
 
-:::note
-
-`getAccounts` only gets data from the legacy account object that was used in `StateService`. As data
-gets migrated off of that account object the response from `getAccounts`, which returns a record
-where the key will be a user's ID and the value being the legacy account object.
-
-:::
+:::warning `firstValueFrom()` and state updates
 
-### Example PRs
+A usage pattern of updating state and then immediately requesting a value through `firstValueFrom()`
+**will not always result in the updated value being returned**. This is because we cannot guarantee
+that the update has taken place before the `firstValueFrom()` executes, in which case the previous
+(cached) value of the observable will be returned.
 
-- [`EnvironmentService`](https://github.com/bitwarden/clients/pull/7621)
-- [Org Keys](https://github.com/bitwarden/clients/pull/7521)
+Use of `firstValueFrom()` should be avoided. If you find yourself trying to use `firstValueFrom()`,
+consider propagating the underlying observable instead of leaving reactivity. :::
```

Comment:
Missing end to the warning block?

Reply:
Oops! Fixed in [`fbd3861` (#549)](https://github.com/bitwarden/contributing-docs/pull/549/commits/fbd3861330cdced948f619c081bea4644cc6ed43).